src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(5);
```

Comment:
Consider defining a constant for the number of figure types instead of using the magic number `5`. For example, `private static final int FIGURE_TYPES_COUNT = 5;`. This will make your code more readable and easier to maintain.